### PR TITLE
Add --json argument for json output

### PR DESCRIPTION
**Pull request by edenriquez**
This PR Fixes https://github.com/samber/git-contrib-graph/issues/2

### CODE REVIEW

File: git-contrib-graph.go [hunk 1]
```diff
@@ -167,7 +177,28 @@ func printAuthors(contribs map[string]map[string]Stats) {
...
+				author,
+				commitCount,
+				additionSum,
+				deletionSum,
+				perDay,
+			)
+		}
+	}
+	if JSON_OUTPUT {
+		// remove last "," character from json
```
Comment (samber) [inline git-contrib-graph.go]: you can do it with `strings.Join(",")` ;-)

File: git-contrib-graph.go [hunk 2]
```diff
@@ -167,7 +177,28 @@ func printAuthors(contribs map[string]map[string]Stats) {
...
+		jsout := string(out)[:len(string(out))-1]
+		fmt.Printf(
+			"\n\n%s\n%s\n\n Json: \n\n%s\n\n",
```
Comment (samber) [inline git-contrib-graph.go]: Could you output results in an array ?

```
[author1, author2, ...]
```

**Review by samber — APPROVED** [review]
thanks @eduenriquez 

this is a nice job 💪 

Could you only display json in output, for that case plz ?

Such as:

```
if JSON_OUTPUT {
   out = `{"author": "foobar@example.com"}`
} else {
   fmt.Printf("Author...")
}
```

(don't forget to update README 😃)

**Comment by samber** [comment]
Can you rebase on master ?

We added unit tests.
Can you check if returned json is parsed correctly by `jq` plz ?

File: git-contrib-graph.go (hunk 1 above)
Comment (edenriquez) [reply git-contrib-graph.go]: sure i'll do that :)

File: git-contrib-graph.go (hunk 2 above)
Comment (edenriquez) [reply git-contrib-graph.go]: :0 sure, that should be better

**Comment by edenriquez** [comment]
@samber  sure i will rebase this night  :) and see testing coverage

**Comment by samber** [comment]
To date, there is no unit test: only a CI with classic checks for go ;)

You're right, we should add some basic tests !

I will create a new issue.

**Comment by edenriquez** [comment]
@samber about first comment 
> Could you only display json in output, for that case plz ?

you mean for each case display only json like
```
################################################################################
################################################################################

{"author": "foobar@example.com"} ...

```

and when there is no --json flag 

```
################################################################################
################################################################################


Author: foobar@example.com ...

```

is it right ?

File: git-contrib-graph.go (hunk 1 above)
Comment (edenriquez) [reply git-contrib-graph.go]: by making this an array i'm no longer need slice this last coma

File: git-contrib-graph.go (hunk 2 above)
Comment (edenriquez) [reply git-contrib-graph.go]: done! :)

File: README.md [hunk 3]
```diff
@@ -106,6 +108,95 @@ Per day:
    2018-09-01 |   0(+)   0(-)
    2018-10-01 | 386(+) 359(-) ++++++++++++++++++++++++++------------------------
  
+```
+## Json Example
+
+Display crontribs per week, with color and even weeks without any commit and json object.
```
Comment (citizen428) [inline README.md]: I suppose `crontribs` is a typo.

Comment (samber) [reply README.md]: yep

**Comment by samber** [comment]
@eduenriquez Please display only JSON: without separator or anything else.

You should be able to parse it with `jq` ;)

**Comment by edenriquez** [comment]
done @samber see README of json flag output, then i cannot figure out how to run jq for my output do you have an example ? 
what i do is 

```./gitlog-jq.sh output.json```

but this happends

```
./gitlog-jq.sh: line 4: jq: command not found
./gitlog-jq.sh: line 4: jq: command not found
./gitlog-jq.sh: line 4: jq: command not found
./gitlog-jq.sh: line 4: jq: command not found
```

so how i can install jq, is a golang dependency or bin script ?

**Comment by samber** [comment]
To install jq => https://stedolan.github.io/jq/download/

Example: 

```
echo '{"toto": 42}' | jq .
```

**Pull request merged**